### PR TITLE
ws-0.2.1-3.1-FetchAPI-ListWhaleSightings

### PR DESCRIPTION
**Pull request by BDoodkorte**
Whale Spotting - Fetch API - List Whale Sightings - End Point Controller

### CODE REVIEW

File: WhaleSpotting/Controllers/WhaleSightingController.cs [hunk 1]
```diff
@@ -27,4 +28,16 @@ public IActionResult GetById([FromRoute] int Id)
             return NotFound();
         }
     }
+
+    [HttpGet("")]
+    public ActionResult<List<WhaleSightingResponse>> ListApprovedSightings()
+    {
+        try{
+             return  _whaleSightingService.ListApprovedSightings();
```
Comment (francescastocco) [inline WhaleSpotting/Controllers/WhaleSightingController.cs:36]: This is tiny but can we reformat this
```suggestion
        try
        {
            return  _whaleSightingService.ListApprovedSightings();
```

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs [hunk 2]
```diff
@@ -32,4 +34,20 @@ public WhaleSighting GetById(int id)
...
+
+    public List<WhaleSightingResponse> ListApprovedSightings()
+    {
+        try
+        {
+            return context.WhaleSightings.Where(ws => (int)ws.ApprovalStatus == 1)
+            .Include(ws => ws.User)
+            .Include(ws => ws.WhaleSpecies)
+            .Select(x => new WhaleSightingResponse(x))
+            .AsEnumerable().ToList();
```
Comment (francescastocco) [inline WhaleSpotting/Repositories/WhaleSightingRepo.cs:46]: Can you do this without the `.AsEnumerable()` statement? Or does it give an error?
```suggestion
            .ToList();
```

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs [hunk 3]
```diff
@@ -32,4 +34,20 @@ public WhaleSighting GetById(int id)
...
     }
+
+    public List<WhaleSightingResponse> ListApprovedSightings()
+    {
+        try
+        {
+            return context.WhaleSightings.Where(ws => (int)ws.ApprovalStatus == 1)
+            .Include(ws => ws.User)
+            .Include(ws => ws.WhaleSpecies)
+            .Select(x => new WhaleSightingResponse(x))
```
Comment (francescastocco) [inline WhaleSpotting/Repositories/WhaleSightingRepo.cs:45]: Do we want to add an `.OrderBy()` statement here?

**Review by francescastocco — CHANGES_REQUESTED** [review]
This is good, just some minor changes :)

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs (hunk 2 above)
Comment (BDoodkorte) [reply WhaleSpotting/Repositories/WhaleSightingRepo.cs:46]: We get a 404 error if we take away .AsEnumerable(), so it's left in.

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs [hunk 4]
```diff
@@ -43,7 +43,8 @@ public List<WhaleSightingResponse> ListApprovedSightings()
             .Include(ws => ws.User)
             .Include(ws => ws.WhaleSpecies)
             .Select(x => new WhaleSightingResponse(x))
-            .AsEnumerable().ToList();
+            .AsEnumerable()
```
Comment (francescastocco) [inline WhaleSpotting/Repositories/WhaleSightingRepo.cs:46]: Thanks, can we remove the `.AsEnumerable()`?

**Review by francescastocco — CHANGES_REQUESTED** [review]
Just another small change

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs (hunk 4 above)
Comment (BDoodkorte) [reply WhaleSpotting/Repositories/WhaleSightingRepo.cs:46]: As explained above, removing .AsEnumerable() leads to errors.

**Review by francescastocco — APPROVED** [review]

**Pull request merged**